inst/include/population_dynamics/population/population.hpp

```diff
@@ -33,15 +33,516 @@
+    std::vector<Type> expected_numbers_at_age;   /*!< Expected values: Numbers at age (thousands?? millions??) */
+    std::vector<Type> catch_numbers_at_age;      /*!< Expected values???: Catch in numbers at age*/
+    std::vector<Type> catch_weight_at_age;       /*!< Expected values???: Weight at age for catch*/
+    std::vector<Type> expected_catch;            /*!< Expected values: Catch*/
+    std::vector<Type> expected_index;            /*!< Expected values: Index (CPUE)*/
+
+    /// recruitment
+    int recruitment_id = -999;                                /*!< id of recruitment model object*/
+    std::shared_ptr<fims::RecruitmentBase<Type> > recruitment; /*!< shared pointer to recruitment module */
+
+    // growth
+    int growth_id = -999;                           /*!< id of growth model object*/
+    std::shared_ptr<fims::GrowthBase<Type> > growth; /*!< shared pointer to growth module */
+
+    // maturity
+    int maturity_id = -999;                             /*!< id of maturity model object*/
+    std::shared_ptr<fims::MaturityBase<Type> > maturity; /*!< shared pointer to maturity module */
+
+    // fleet
+    int fleet_id = -999;                                    /*!< id of fleet model object*/
+    std::vector<std::shared_ptr<fims::Fleet<Type> > > fleets; /*!< shared pointer to fleet module */
+
+    // this -> means you're referring to a class member (member of self)
+
+    // std::vector<std::shared_ptr<fims::Fleet<Type>> > surveys;
+    Population()
+    {
+      this->id = Population::id_g++;
+    }
+
+    /**
+     * @brief Initialize values. Called once at the start of model run.
+     *
+     * @param nyears number of years in the population
+     * @param nseasons number of seasons in the population
+     * @param nages number of ages in the population
+     */
+    void Initialize(int nyears, int nseasons, int nages)
+    {
+      // size all the vectors to length of nages
+      nfleets = fleets.size();
+      ages.resize(nages);
+      catch_numbers_at_age.resize(nyears * nages * nfleets);
+      mortality_F.resize(nyears * nages);
+      mortality_Z.resize(nyears * nages);
+      // proportion_mature_at_age.resize(nyears * nages);
+      proportion_mature_at_age.resize((nyears + 1) * nages);
+      weight_at_age.resize(nages);
+      catch_weight_at_age.resize(nyears * nages * nfleets);
+      unfished_numbers_at_age.resize((nyears + 1) * nages);
+      numbers_at_age.resize((nyears + 1) * nages);
+      expected_catch.resize(nyears * nfleets);
+      expected_index.resize(nyears * nfleets);
+      biomass.resize((nyears + 1));
+      unfished_spawning_biomass.resize((nyears + 1));
+      spawning_biomass.resize((nyears + 1));
+      log_naa.resize(nages);
+      log_M.resize(nyears * nages);
+      naa.resize(nages);
+
+      M.resize(nyears * nages);
+
+      for (size_t fleet_ = 0; fleet_ < this->nfleets; fleet_++)
+      {
+        this->fleets[fleet_]->Initialize(nyears, nages);   
```

Comment:
fleets should not be initialized here because it is initialized in[ information.hpp](https://github.com/NOAA-FIMS/FIMS/blob/abd86cb21b7cf50d2b5c7e5c652e16dc9c91a218/inst/include/common/information.hpp#L141), is this correct @msupernaw?